### PR TITLE
Introduce testfeed details mode for ConsoleLauncher

### LINKED ISSUE #2381: Provide a concise test feed details mode for the ConsoleLauncher
The ability to have a good report (tree) does not prevent to have some real time hints about executed tests.
Currently, these requirements are not fulfilled OOTB by the console launcher.
Would be great to have a listener logging each test being executed (a bit like surefire does "Running X") + the global report at the end.

### PR DESCRIPTION
**Pull request by marcphilipp**
## Overview

Resolves #2381.

---

I hereby agree to the terms of the [JUnit Contributor License Agreement](https://github.com/junit-team/junit5/blob/002a0052926ddee57cf90580fa49bc37e5a72427/CONTRIBUTING.md#junit-contributor-license-agreement).

---

### Definition of Done

- [x] There are no TODOs left in the code
- [x] Method [preconditions](https://junit.org/junit5/docs/snapshot/api/org.junit.platform.commons/org/junit/platform/commons/util/Preconditions.html) are checked and documented in the method's Javadoc
- [x] [Coding conventions](https://github.com/junit-team/junit5/blob/HEAD/CONTRIBUTING.md#coding-conventions) (e.g. for logging) have been followed
- [x] Change is covered by [automated tests](https://github.com/junit-team/junit5/blob/HEAD/CONTRIBUTING.md#tests) including corner cases, errors, and exception handling
- [x] Public API has [Javadoc](https://github.com/junit-team/junit5/blob/HEAD/CONTRIBUTING.md#javadoc) and [`@API` annotations](https://apiguardian-team.github.io/apiguardian/docs/current/api/org/apiguardian/api/API.html)
- [x] Change is documented in the [User Guide](https://junit.org/junit5/docs/snapshot/user-guide/) and [Release Notes](https://junit.org/junit5/docs/snapshot/user-guide/#release-notes)
- [x] All [continuous integration builds](https://github.com/junit-team/junit5#continuous-integration-builds) pass

### CODE REVIEW

File: platform-tests/src/test/java/org/junit/platform/console/tasks/TestFeedPrintingListenerTests.java [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+	}
+
+	@Test
+	public void testExecutionSkipped() {
+		listener.executionSkipped(testIdentifier, "Test disabled");
+		assertLinesMatch( //
+			"""
+					Demo Engine > %c ool test :: SKIPPED
+					\tReason: Test disabled
+					""".lines(), //
```
Comment (sormuras) [inline platform-tests/src/test/java/org/junit/platform/console/tasks/TestFeedPrintingListenerTests.java:60]: Indentation is off

File: junit-platform-console/src/main/java/org/junit/platform/console/tasks/DetailsPrintingListener.java [hunk 2]
```diff
@@ -10,18 +10,18 @@
...
 
-import org.apiguardian.api.API;
 import org.junit.platform.launcher.TestExecutionListener;
 import org.junit.platform.launcher.TestPlan;
 
-/**
- * @since 1.9
- */
-@API(status = EXPERIMENTAL, since = "1.9")
-public interface DetailsPrintingListener extends TestExecutionListener {
```
Comment (sormuras) [inline junit-platform-console/src/main/java/org/junit/platform/console/tasks/DetailsPrintingListener.java:23]: Making this package-private is good.

**Review by sormuras — APPROVED** [review]
Looks good overall.
Nit: should test reporter messages be printed?

**Comment by marcphilipp** [comment]
I was also wondering about that. Are they really needed for a concise feed, though?

File: platform-tests/src/test/java/org/junit/platform/console/tasks/TestFeedPrintingListenerTests.java (hunk 1 above)
Comment (marcphilipp) [reply platform-tests/src/test/java/org/junit/platform/console/tasks/TestFeedPrintingListenerTests.java:60]: This is what Spotless/the Eclipse formatter does.

Comment (sormuras) [reply platform-tests/src/test/java/org/junit/platform/console/tasks/TestFeedPrintingListenerTests.java:60]: Then they are no good team. :P

**Comment by sormuras** [comment]
> I was also wondering about that. Are they really needed for a concise feed, though?

No. Probably not. Let's leave them out.

**Pull request merged**